modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc

```diff
@@ -449,22 +449,32 @@ bool HybridAStar::CombinePathAndSpeedProfile(
   result->phi.clear();
   result->v.clear();
   result->a.clear();
+  result->steer.clear();
   result->accumulated_s.clear();
 
   if (discretized_path.empty()) {
-    AWARN << "path data is empty";
+    AERROR << "path data is empty";
     return false;
   }
 
-  for (double cur_rel_time = 0.0; cur_rel_time < speed_data.TotalTime();
+  const double kEpsilon = 1e-6;
+
+  for (double cur_rel_time = 0.0;
+       cur_rel_time < speed_data.TotalTime() + kEpsilon;
```

Comment:
Calculate the upper bound once and keep the result in variable, to make it shorter and more efficient.